### PR TITLE
WIP: C++11 support for std::make_unique

### PR DESCRIPTION
**Pull request by atanisoft**
Since arduino-esp32 defaults to C++11 (due to ESP-IDF using the same) the usage of std::make_unique caused build failures for @RobertPHeller (as noted on groups.io). After a bit of digging into how to detect that we are building on an older version of the C++ standard I was able to include a version of std::make_unique as submitted for C++ standards as a header include.

Also fixed an issue with Esp32WiFiManager compilation on arduino-esp32 v1.0.5 (rc1 and rc2) since it is picking up ESP-IDF v3.3.x which is now defining `ESP_IDF_VERSION_MAJOR` (and related version defines).

### CODE REVIEW

**Comment by RobertPHeller** [comment]
i am getting this error:

/home/heller/Arduino/libraries/OpenMRN/src/freertos_drivers/esp32/Esp32WiFiManager.cpp:54:43: error: missing binary operator before token "("
     ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(4,0,0)
                                                                                   ^

**Comment by atanisoft** [comment]
Hmm, it looks like it doesn't like the split preprocessor directives, can you combine it onto one line? Not sure why I didn't see this on my compiles.

**Comment by RobertPHeller** [comment]
At Sun, 22 Nov 2020 07:28:52 -0800 bakerstu/openmrn <reply@reply.github.com> wrote:

> 

> Hmm, it looks like it doesn't like the split preprocessor directives, can
> you combine it onto one line? Not sure why I didn't see this on my compiles.

that's not it.  same error.


> 

-- 
Robert Heller             -- Cell: 413-658-7953 GV: 978-633-5364
Deepwoods Software        -- Custom Software Services
http://www.deepsoft.com/  -- Linux Administration Services
heller@deepsoft.com       -- Webhosting Services

**Comment by atanisoft** [comment]
Ok let me do some more digging as it seems to be esp-idf version dependent, specifically pre-3.3 builds. Arduino-esp32 1.0.5 is stable and nearly ready for release but it will take time for people to move up to it.

**Comment by RobertPHeller** [comment]
At Sun, 22 Nov 2020 07:39:32 -0800 bakerstu/openmrn <reply@reply.github.com> wrote:

> 

> Ok let me do some more digging as it seems to be esp-idf version dependent,
> specifically pre-3.3 builds. Arduino-esp32 1.0.5 is stable and nearly ready
> for release but it will take time for people to move up to it.

I am using the "stock" Arduino-esp32 1.0.4 on a Raspberry Pi 2 (armv7l):

Using built-in specs.
COLLECT_GCC=/home/heller/.arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/1.22.0-80-g6c4433a-5.2.0/bin/xtensa-esp32-elf-c++
COLLECT_LTO_WRAPPER=/home/heller/.arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/1.22.0-80-g6c4433a-5.2.0/bin/../libexec/gcc/xtensa-esp32-elf/5.2.0/lto-wrapper
Target: xtensa-esp32-elf
Configured with: /builds/idf/crosstool-NG/.build/src/gcc-5.2.0/configure --build=x86_64-build_pc-linux-gnu --host=arm-host_unknown-linux-gnueabi --target=xtensa-esp32-elf --prefix=/builds/idf/crosstool-NG/builds/xtensa-esp32-elf --with-local-prefix=/builds/idf/crosstool-NG/builds/xtensa-esp32-elf/xtensa-esp32-elf/sysroot --with-sysroot=/builds/idf/crosstool-NG/builds/xtensa-esp32-elf/xtensa-esp32-elf/sysroot --with-newlib --enable-threads=no --disable-shared --with-pkgversion='crosstool-NG crosstool-ng-1.22.0-87-gb57bad32' --disable-__cxa_atexit --enable-cxx-flags='-fno-rtti -ffunction-sections' --with-gmp=/builds/idf/crosstool-NG/.build/xtensa-esp32-elf/buildtools/complibs-host --with-mpfr=/builds/idf/crosstool-NG/.build/xtensa-esp32-elf/buildtools/complibs-host --with-mpc=/builds/idf/crosstool-NG/.build/xtensa-esp32-elf/buildtools/complibs-host --with-isl=/builds/idf/crosstool-NG/.build/xtensa-esp32-elf/buildtools/complibs-host --with-cloog=/builds/idf/crosstool-NG/.build/xtensa-es
 p32-elf/buildtools/complibs-host --with-libelf=/builds/idf/crosstool-NG/.build/xtensa-esp32-elf/buildtools/complibs-host --enable-lto --with-host-libstdcxx='-static-libgcc -Wl,-Bstatic,-lstdc++ -lm' --enable-target-optspace --without-long-double-128 --disable-libgomp --disable-libmudflap --disable-libssp --disable-libquadmath --disable-libquadmath-support --disable-nls --enable-languages=c,c++ --disable-libstdcxx-verbose --enable-threads=posix --enable-gcov-custom-rtio
Thread model: posix
gcc version 5.2.0 (crosstool-NG crosstool-ng-1.22.0-87-gb57bad32) 


> 

-- 
Robert Heller             -- Cell: 413-658-7953 GV: 978-633-5364
Deepwoods Software        -- Custom Software Services
http://www.deepsoft.com/  -- Linux Administration Services
heller@deepsoft.com       -- Webhosting Services

File: src/freertos_drivers/esp32/Esp32WiFiManager.cxx [hunk 1]
```diff
@@ -50,15 +50,16 @@
 
 // ESP-IDF v4+ has a slightly different directory structure to previous
 // versions.
-#ifdef ESP_IDF_VERSION_MAJOR
```
Comment (balazsracz) [inline src/freertos_drivers/esp32/Esp32WiFiManager.cxx:53]: plesae putthis change into a separate PR as it has nothing to do with the make unique.

Comment (atanisoft) [reply src/freertos_drivers/esp32/Esp32WiFiManager.cxx:53]: Ok, will split but it won't compile without it very soon.

**Pull request closed without merging**